pkg/registry/horizontalpodautoscaler/etcd/etcd_test.go

```diff
@@ -0,0 +1,231 @@
+			return false
+		}
+		return fakeEtcdClient.Data[key].R.Node.TTL == 30
+	}
+	test.TestDeleteNoGraceful(createFn, gracefulSetFn)
+}
+
+func TestEtcdGet(t *testing.T) {
+	ctx := api.NewDefaultContext()
+	registry, fakeClient, _ := newStorage(t)
+	autoscaler := validNewHorizontalPodAutoscaler("foo3")
+	name := autoscaler.Name
+	key, _ := registry.KeyFunc(ctx, name)
+	key = etcdtest.AddPrefix(key)
+	fakeClient.Set(key, runtime.EncodeOrDie(testapi.Codec(), autoscaler), 0)
+	response, err := fakeClient.Get(key, false, false)
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+	var autoscalerOut expapi.HorizontalPodAutoscaler
+	err = testapi.Codec().DecodeInto([]byte(response.Node.Value), &autoscalerOut)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	obj, err := registry.Get(ctx, name)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	got := obj.(*expapi.HorizontalPodAutoscaler)
+	autoscaler.ObjectMeta.ResourceVersion = got.ObjectMeta.ResourceVersion
+	if e, a := autoscaler, got; !api.Semantic.DeepEqual(*e, *a) {
+		t.Errorf("Unexpected autoscaler: %#v, expected %#v", e, a)
+	}
+}
+
+func TestEmptyList(t *testing.T) {
+	ctx := api.NewDefaultContext()
+	registry, fakeClient, _ := newStorage(t)
+	fakeClient.ChangeIndex = 1
+	key := registry.KeyRootFunc(ctx)
+	key = etcdtest.AddPrefix(key)
+	fakeClient.Data[key] = tools.EtcdResponseWithError{
+		R: &etcd.Response{},
+		E: fakeClient.NewError(tools.EtcdErrorCodeNotFound),
+	}
+	autoscalerList, err := registry.List(ctx, labels.Everything(), fields.Everything())
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if len(autoscalerList.(*expapi.HorizontalPodAutoscalerList).Items) != 0 {
+		t.Errorf("Unexpected non-zero autoscaler list: %#v", autoscalerList)
+	}
+	if autoscalerList.(*expapi.HorizontalPodAutoscalerList).ResourceVersion != "1" {
+		t.Errorf("Unexpected resource version: %#v", autoscalerList)
+	}
+}
+
+func TestList(t *testing.T) {
+	ctx := api.NewDefaultContext()
+	registry, fakeClient, _ := newStorage(t)
+	fakeClient.ChangeIndex = 1
+	key := registry.KeyRootFunc(ctx)
+	key = etcdtest.AddPrefix(key)
+	fakeClient.Data[key] = tools.EtcdResponseWithError{
+		R: &etcd.Response{
+			Node: &etcd.Node{
+				Nodes: []*etcd.Node{
+					{
+						Value: runtime.EncodeOrDie(testapi.Codec(), &expapi.HorizontalPodAutoscaler{
+							ObjectMeta: api.ObjectMeta{Name: "foo"},
+						}),
+					},
+					{
+						Value: runtime.EncodeOrDie(testapi.Codec(), &expapi.HorizontalPodAutoscaler{
+							ObjectMeta: api.ObjectMeta{Name: "bar"},
+						}),
+					},
+				},
+			},
+		},
+	}
+	obj, err := registry.List(ctx, labels.Everything(), fields.Everything())
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+	autoscalerList := obj.(*expapi.HorizontalPodAutoscalerList)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	if len(autoscalerList.Items) != 2 {
+		t.Errorf("Unexpected persistentVolume list: %#v", autoscalerList)
```

Comment:
I assume all of the error messages that say persistentVolume should say HorizontalPodAutoscaler (or autoscaler)?